ludwig/trainers/trainer_lightgbm.py

```diff
@@ -945,64 +950,76 @@ def __init__(
     def get_schema_cls() -> BaseTrainerConfig:
         return GBMTrainerConfig
 
-    def train_step(
+    def _train_loop(
         self,
         params: Dict[str, Any],
-        lgb_train: "RayDMatrix",  # noqa: F821
-        eval_sets: List["RayDMatrix"],  # noqa: F821
+        lgb_train: lgb.Dataset,
+        eval_sets: List[lgb.Dataset],
```

Comment:
Why are we cutting support for `RayDMatrix`? Is it just because of the index error?

Reply:
Good catch! this is a mistake because of some copy pasta from LightGBMTrainer which uses lgb.Dataset instead of RayDMatrix. Just updated!